src/compile/repeat.ts

```diff
@@ -32,7 +32,7 @@ export function replaceRepeaterInEncoding(encoding: Encoding<Field>, repeater: R
   return replaceRepeater(encoding, repeater);
 }
 
-type EncodingOrFacet<F> = Encoding<F> | Facet<F>;
+type EncodingOrFacet<F> = Encoding<F> & Facet<F>;
```

Comment:
We want an intersection type, which is **not** an intersection of the fields of those types.

```ts
interface A {
  a: number;
  b: number;
}

interface B {
  b: number;
  c: number;
}

const o: A & B;
```

Also, see https://twitter.com/domoritz/status/858899054507442176